### PR TITLE
LOON-416: Add lastConfirmedDate, refactor code

### PR DESCRIPTION
**Pull request by d35h**

### CODE REVIEW

File: backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt [hunk 1]
```diff
@@ -53,10 +53,12 @@ class PreventionService(
                 }
                 ?.sortedBy { it.date } ?: listOf(ExaminationRecord())
 
-            var count = 0
```
[the remark this answers is not in the document]
Comment (rposkocil) [reply backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt:56]: You are right. Your solution is more simple then the previous code. Great catch! :-)

File: backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt [hunk 2]
```diff
@@ -53,10 +53,12 @@ class PreventionService(
...
-                count = examsOfType.map { it.status == ExaminationStatusDto.CONFIRMED }.size
-            }
+            val confirmedExamsOfCurrentType = examsOfType?.filter { it.status == ExaminationStatusDto.CONFIRMED }
+            // 1) Filter all the confirmed records
+            // 2) Map all non-nullable lastExamination records
+            // 3) Find the largest or return null if the list is empty
+            val lastConfirmedDate = confirmedExamsOfCurrentType?.mapNotNull(ExaminationRecord::date)?.maxOrNull()
+            val totalCountOfConfirmedExams = confirmedExamsOfCurrentType?.size ?: 0
 
             PreventionStatusDto(
```
Comment (rposkocil) [inline backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt:63]: I know that it is a minority - I would also rename 'date' field to something more clear like 'plannedDate' to more differentiate the dates.

File: backend/src/main/resources/doc/openapi.json [hunk 3]
```diff
@@ -1148,6 +1148,10 @@
                     },
                     "count": {
                         "type": "integer"
+                    },
+                    "lastConfirmedDate": {
```
Comment (rposkocil) [inline backend/src/main/resources/doc/openapi.json:1152]: I suppose that you have edited it manually. This openApi documentation is exported from stoplight.io where should be edited, too.  It is placed here to generate DTO objects in this project. The truth between BE and FE is this OpenAPI document located in the loono-api project. The mobile app generates the API objects from this file on its side.

File: backend/src/test/kotlin/cz/loono/backend/api/service/PreventionServiceTest.kt [hunk 4]
```diff
@@ -38,28 +38,44 @@ class PreventionServiceTest {
         whenever(accountRepository.findByUid(uuid)).thenReturn(account)
         whenever(examinationRecordRepository.findAllByAccountOrderByDateDesc(account)).thenReturn(
             setOf(
```
[the remark this answers is not in the document]
Comment (rposkocil) [reply backend/src/test/kotlin/cz/loono/backend/api/service/PreventionServiceTest.kt:40]: Interesting, I hope that I have caught your idea. I have extended these tests by some fields and changed some default values. The test reacted on any change of any field not just Id field. I suppose that this code is generated by some Hibernate equals generator or IDE plugin.

**Review by rposkocil — COMMENTED** [review]
LGTM. You have find your way through quickly.

File: backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt (hunk 2 above)
Comment (d35h) [reply backend/src/main/kotlin/cz/loono/backend/api/service/PreventionService.kt:63]: Ok, done in c8a37fa

File: backend/src/main/resources/doc/openapi.json (hunk 3 above)
Comment (d35h) [reply backend/src/main/resources/doc/openapi.json:1152]: Ok, is there a way I can get an access there?

**Comment by d35h** [comment]
Also, I refactored code a bit, to make it more Kotlin-like

File: backend/src/main/resources/doc/openapi.json (hunk 3 above)
Comment (d35h) [reply backend/src/main/resources/doc/openapi.json:1152]: Bump :)

Comment (d35h) [reply backend/src/main/resources/doc/openapi.json:1152]: @rposkocil done in https://github.com/cesko-digital/loono-api/pull/2

**Review by martin-ladecky — APPROVED** [review]
Lgtm

**Review by rposkocil — APPROVED** [review]

**Pull request merged**